vowpalwabbit/options.h

```diff
@@ -124,12 +158,38 @@ struct typed_option : base_option
 
   bool value_supplied() const { return m_value.get() != nullptr; }
 
+  template <typename U>
+  std::string invalid_choice_error(const U&)
+  {
+    return "";
+  }
+  std::string invalid_choice_error(const std::string& value)
+  {
+    std::ostringstream err_msg;
+    err_msg << fmt::format("Error: '{}' is not a valid choice for option --{}", value, m_name);

```

Comment:
I think when throwing you already print this information, check stderr file for duplication of same message

Reply:
This part generates the message, there's only one `THROW` for this message on 362 in options.h, not sure why its showing up twice